### PR TITLE
Nest view under search

### PR DESCRIPTION
**Pull request by EricSommerhalder**

### CODE REVIEW

File: src/SearchComponents/SearchField.svelte [hunk 1]
```diff
@@ -1,21 +1,45 @@
...
     */
     export function getString() {
-
-        return getPropString(prop);
+        let toReturn = getPropStringHelper(prop, parent);
+        if (children.length > 0){
+            for (const child of children){
+                toReturn += child.getString();
+            }
+        }
```
Comment (Vijeinath) [inline src/SearchComponents/SearchField.svelte]: Is this if clause necessary? As far as I understood the for loop only starts if the children array has as least on element.

File: src/SearchComponents/SearchField.svelte [hunk 2]
```diff
@@ -1,21 +1,45 @@
...
+        let toReturn = '';
+        if (isEmpty()){
+            toReturn += 'OPTIONAL {\n';
+        }
+        toReturn += getPropStringHelper(prop, parent);
+        if (children.length > 0){
+            for (const child of children){
+                toReturn += child.getOptionalString();
+            }
+        }
```
Comment (Vijeinath) [inline src/SearchComponents/SearchField.svelte]: Is this if clause necessary? As far as I understood the for loop only starts if the children array has as least on element.

File: src/SearchComponents/SearchField.svelte [hunk 3]
```diff
@@ -1,21 +1,45 @@
...
-        return getPropString(prop);
+        let toReturn = getPropStringHelper(prop, parent);
+        if (children.length > 0){
+            for (const child of children){
+                toReturn += child.getString();
+            }
+        }
+        return toReturn;
+    }
+    export function getOptionalString(){
```
Comment (Vijeinath) [inline src/SearchComponents/SearchField.svelte]: For better readability it is preferable that a empty line should be inserted between functions

File: src/SearchComponents/SimilarSearch.svelte [hunk 4]
```diff
@@ -2,55 +2,72 @@
...
+                    prop["linkQueue"] = linkQueue;
+                }
                 searchInfo.push(prop);
+                console.log(searchInfo);
             }
-
         }
-        await search();
+    }
+    function getStringForProp(prop){
```
Comment (Vijeinath) [inline src/SearchComponents/SimilarSearch.svelte]: For better readability it is preferable that a empty line should be inserted between functions

File: src/SearchComponents/SimilarSearch.svelte [hunk 5]
```diff
@@ -2,55 +2,72 @@
...
-            while (linkStack.length > 0){
-                curr = linkStack.pop();
-                linkedIri = linkedRes[ontology + ':' + curr + 'Value']['knora-api:linkValueHasTarget']['@id'];
-                linkedRes = await getResByIri(linkedIri, $token);
+        for (const prop of props) {
+            await getPropValue(prop, data);
+        }
+        await search();
+    }
+    async function getPropValue(prop, data, linkQueue = []){
```
Comment (Vijeinath) [inline src/SearchComponents/SimilarSearch.svelte]: Empty line between functions

File: src/SearchComponents/SearchUtility.js [hunk 6]
```diff
@@ -4,29 +4,27 @@ import {get} from 'svelte/store';
 const jVal = get(json);
 let ontology = jVal['DSP']['Ontology'];
 let shortName = jVal['DSP']['ShortName'];
+
 export function getPropString(prop){
+    return getPropStringHelper(prop, "?mainres");
+}
+export function getPropStringHelper(prop, parent){
```
Comment (Vijeinath) [inline src/SearchComponents/SearchUtility.js]: Empty line between functions

File: src/SearchComponents/SearchUtility.js [hunk 7]
```diff
@@ -4,29 +4,27 @@ import {get} from 'svelte/store';
...
         }
-        parent = '?' + currProp['propName'];
+
+    } else {
+        toReturn += parent + ' ' + shortName + ':' + prop['propName'] + ' ?' + prop['propName'] + ' .\n';
     }
+
     return toReturn;
 }
 export function getFilterByNameValAndObj(propName, value, obj, dateDepth=''){
```
Comment (Vijeinath) [inline src/SearchComponents/SearchUtility.js]: Empty line between functions

File: src/SearchComponents/SimilarSearch.svelte [hunk 8]
```diff
@@ -2,55 +2,72 @@
...
+                curr = temp.shift();
+                toReturn += parent + ' ' + shortName + ':' + curr + ' ?' + curr + ' .\n';
+                parent = '?' + curr;
+            }
+        }
+        toReturn += getPropStringHelper(prop, parent);
+        console.log(prop);
+        return toReturn;
     }
     async function search(){
```
Comment (Vijeinath) [inline src/SearchComponents/SimilarSearch.svelte]: Empty line between functions

**Review by Vijeinath — CHANGES_REQUESTED** [review]
There are some minor things to change.

File: src/SearchComponents/SearchField.svelte (hunk 1 above)
Comment (EricSommerhalder) [reply src/SearchComponents/SearchField.svelte]: agreed

File: src/SearchComponents/SearchField.svelte (hunk 2 above)
Comment (EricSommerhalder) [reply src/SearchComponents/SearchField.svelte]: agreed

File: src/SearchComponents/SearchField.svelte (hunk 3 above)
Comment (EricSommerhalder) [reply src/SearchComponents/SearchField.svelte]: Done

File: src/SearchComponents/SimilarSearch.svelte (hunk 4 above)
Comment (EricSommerhalder) [reply src/SearchComponents/SimilarSearch.svelte]: Done

File: src/SearchComponents/SimilarSearch.svelte (hunk 5 above)
Comment (EricSommerhalder) [reply src/SearchComponents/SimilarSearch.svelte]: Done

**Pull request merged**